pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test: TestSQLStatsRegions failed

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18439436?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18439436?buildTab=artifacts#/) on master @ [31e5601ca0b65266ae0de37eae2aa0fef0cd62b3](https://github.com/cockroachdb/cockroach/commits/31e5601ca0b65266ae0de37eae2aa0fef0cd62b3):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions3674978010
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:196: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions3674978010
--- FAIL: TestSQLStatsRegions (78.76s)
=== RUN   TestSQLStatsRegions/system_tenant
    sql_stats_test.go:191: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:191
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:147
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-east1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,2 @@
        	            	-([]string) (len=3) {
        	            	- (string) (len=15) "gcp-us-central1",
        	            	+([]string) (len=2) {
        	            	  (string) (len=12) "gcp-us-east1",
        	Test:       	TestSQLStatsRegions/system_tenant
    --- FAIL: TestSQLStatsRegions/system_tenant (2.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46334

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18476972?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18476972?buildTab=artifacts#/) on master @ [91a08e7c6fa917d103ea1fa1d806fec3b6162bb1](https://github.com/cockroachdb/cockroach/commits/91a08e7c6fa917d103ea1fa1d806fec3b6162bb1):


```
=== RUN   TestSQLStatsRegions
I250113 13:23:21.072181 1 (gostd) rand.go:246  [-] 1  random seed: 1296582832080311763
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions3133442671
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:196: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions3133442671
--- FAIL: TestSQLStatsRegions (80.51s)
=== RUN   TestSQLStatsRegions/system_tenant
    sql_stats_test.go:191: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:191
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:147
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-central1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,3 @@
        	            	-([]string) (len=3) {
        	            	+([]string) (len=2) {
        	            	  (string) (len=15) "gcp-us-central1",
        	            	- (string) (len=12) "gcp-us-east1",
        	            	  (string) (len=12) "gcp-us-west1"
        	Test:       	TestSQLStatsRegions/system_tenant
    --- FAIL: TestSQLStatsRegions/system_tenant (0.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18490364?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18490364?buildTab=artifacts#/) on master @ [097438ac38e411b0fde101ebcae5cf97a798d1db](https://github.com/cockroachdb/cockroach/commits/097438ac38e411b0fde101ebcae5cf97a798d1db):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions186605964
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:196: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions186605964
--- FAIL: TestSQLStatsRegions (34.12s)
=== RUN   TestSQLStatsRegions/secondary_tenant
    sql_stats_test.go:191: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:191
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:147
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-central1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,3 @@
        	            	-([]string) (len=3) {
        	            	+([]string) (len=2) {
        	            	  (string) (len=15) "gcp-us-central1",
        	            	- (string) (len=12) "gcp-us-east1",
        	            	  (string) (len=12) "gcp-us-west1"
        	Test:       	TestSQLStatsRegions/secondary_tenant
    --- FAIL: TestSQLStatsRegions/secondary_tenant (22.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://tanzanite.cluster.engflow.com/invocations/default/c10b4ef4-6c60-44f5-83a7-801c55cd8657?testReportRun=22&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3Fsc3RhdHNjY2w6c3Fsc3RhdHNjY2xfdGVzdA==) on master @ [097438ac38e411b0fde101ebcae5cf97a798d1db](https://github.com/cockroachdb/cockroach/commits/097438ac38e411b0fde101ebcae5cf97a798d1db):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestSQLStatsRegions3003503425
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:196: -- test log scope end --
test logs left over in: outputs.zip/logTestSQLStatsRegions3003503425
--- FAIL: TestSQLStatsRegions (114.89s)
=== RUN   TestSQLStatsRegions/secondary_tenant
    sql_stats_test.go:191: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:191
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:147
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-central1", "gcp-us-east1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,5 +1,4 @@
        	            	-([]string) (len=3) {
        	            	+([]string) (len=2) {
        	            	  (string) (len=15) "gcp-us-central1",
        	            	- (string) (len=12) "gcp-us-east1",
        	            	- (string) (len=12) "gcp-us-west1"
        	            	+ (string) (len=12) "gcp-us-east1"
        	            	 }
        	Test:       	TestSQLStatsRegions/secondary_tenant
    --- FAIL: TestSQLStatsRegions/secondary_tenant (8.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18503712?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18503712?buildTab=artifacts#/) on master @ [f1a746cf3e2eb39b6cc68c036a6e2a24d15b70e1](https://github.com/cockroachdb/cockroach/commits/f1a746cf3e2eb39b6cc68c036a6e2a24d15b70e1):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions1674159824
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:196: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions1674159824
--- FAIL: TestSQLStatsRegions (87.17s)
=== RUN   TestSQLStatsRegions/system_tenant
    sql_stats_test.go:191: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:191
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:147
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-east1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,2 @@
        	            	-([]string) (len=3) {
        	            	- (string) (len=15) "gcp-us-central1",
        	            	+([]string) (len=2) {
        	            	  (string) (len=12) "gcp-us-east1",
        	Test:       	TestSQLStatsRegions/system_tenant
    --- FAIL: TestSQLStatsRegions/system_tenant (1.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18511748?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18511748?buildTab=artifacts#/) on master @ [fd919f9a94b6ebb9f9691429cdb3587ab9ae7e33](https://github.com/cockroachdb/cockroach/commits/fd919f9a94b6ebb9f9691429cdb3587ab9ae7e33):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions74682324
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:196: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1ffa8747a1250ee574ba4234d629ddbd/logTestSQLStatsRegions74682324
--- FAIL: TestSQLStatsRegions (37.52s)
=== RUN   TestSQLStatsRegions/system_tenant
    sql_stats_test.go:191: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:191
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:147
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-east1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,2 @@
        	            	-([]string) (len=3) {
        	            	- (string) (len=15) "gcp-us-central1",
        	            	+([]string) (len=2) {
        	            	  (string) (len=12) "gcp-us-east1",
        	Test:       	TestSQLStatsRegions/system_tenant
    --- FAIL: TestSQLStatsRegions/system_tenant (5.54s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go
+++ b/pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go
@@ -150,7 +150,10 @@ func TestSQLStatsRegions(t *testing.T) {
 				if err != nil {
 					return err
 				}
-
+				_, err = db.DB.ExecContext(ctx, `SET distsql = on;`)
+				if err != nil {
+					return err
+				}
 				// Use EXPLAIN ANALYSE (DISTSQL) to get the accurate list of nodes.
 				explainInfo, err := db.DB.QueryContext(ctx, `EXPLAIN ANALYSE (DISTSQL) SELECT * FROM test`)
 				if err != nil {

